lib/ClangImporter/ClangModuleDependencyScanner.cpp

```diff
@@ -204,14 +204,29 @@ static void recordModuleDependencies(
     for (const auto &fileDep : clangModuleDep.FileDeps) {
       fileDeps.push_back(fileDep.getKey().str());
     }
-
+    std::vector<std::string> allArgs = Impl.ClangArgs;
+    for(auto arg: clangModuleDep.NonPathCommandLine) {
+      allArgs.push_back("-Xclang");
+      allArgs.push_back(arg);
+    }
```

Comment:
The arguments in `NonPathCommandLine` are intended to be directly appended to a `-cc1` command line, not to the driver.